dcpy/test/connectors/edm/recipes/test_recipes.py

```diff
@@ -69,3 +80,48 @@ def test_serializing_and_deserializing(self):
         """Deserializing python models is a minefield."""
         lock_file = recipes.plan(RECIPE_NO_DEFAULTS_PATH)
         recipes.recipe_from_yaml(lock_file)
+
+
+_test_df = df([["1", "2"], ["3", "4"]], columns=["a", "b"])
+
+
+def _mock_preprocessor(name, df):
+    df["name"] = name
+    return df
+
+
+def _mock_fetch_dataset(ds, _):
+    out_path = TEMP_DATA_PATH / f"{ds.name}.csv"
+    _test_df.to_csv(out_path, index=False)
+    return out_path
+
+
+@pytest.mark.usefixtures("file_setup_teardown")
+class TestImportDatasets(TestCase):
+    # TODO: move this functionality into an integration test with an actual database
+
+    def set_mocks(self):
+        # Mock out fetch_dataset to download a mocked csv
+        recipes.fetch_dataset = MagicMock(side_effect=_mock_fetch_dataset)
+        # Mocking out a preprocessor for our datasets. Sticking it on `dcpy` for ease
+        dcpy.preproc = MagicMock(side_effect=_mock_preprocessor)  # type: ignore
+
+    def test_import_dataset(self):
+        self.set_mocks()
+        pg_mock = MagicMock()
+        ds = recipes.Dataset(
+            name="test",
+            version="1",
+            import_as="new_table_name",
+            file_type=recipes.DatasetType.csv,
+            preprocessor=recipes.DataPreprocessor(module="dcpy", function="preproc"),
+        )
+        recipes.import_dataset(ds, pg_mock)
+
+        # Verify the values that PostgresClient.insert_dataframe was called with
+        (
+            df_inserted_actual,
+            table_insert_name_actual,
+        ) = pg_mock.insert_dataframe.call_args_list[0][0]
+        assert df_inserted_actual.equals(_mock_preprocessor(ds.name, _test_df))
+        assert table_insert_name_actual == ds.import_as
```

Comment:
cool I've never seen `call_args_list` before!

would it be nice to construct `_test_df` with the column `name` already in it or is it nicer to be more dynamic and use `ds.name` to build the "expected" dataframe?

Reply:
Yeah, I haven't used it before either. I've typically just used the `assert_called_with` style methods to check args, but dataframes are weird about equality. 

Re: your question, the point of the test is largely to check that this mock preprocessor that we've bolted onto dcpy actually gets called. Since we have access to that method here, makes sense to just use it to verify outputs. The actual transformation is pretty negligable - just chose it because we do something similar in facdb, IIRC